<a href="https://colab.research.google.com/github/pacomm5/PLANTILLAS2/blob/main/U382_FECHA_APERTURA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# SUBIMOS PROVISIONALMENTE EL ARCHIVO A GOOGLE COLAB. 
# El archivo esta aquí C:\Users\FJMartin\OneDrive - JARMAUTO\Documentos\2018\POWER BI INFORMES\DATOS MAESTROS\MACHINE LEARNING
# y es el que esta debajo CONSULTA1_AGR_F_APERTURA  es el u382 agrupado por fecha de apertura


In [1]:
import pandas as pd

- esto abriendo una consulta dentro de un excel.
- lo he subido a colab y luego lo he abierto. TArda un poco en subir.
- tengo que poner el nombre de la pestaña. 
- ojo tarda un poco en subir el archivo a colab. un archivo de 34mb ha tardado 1 minuto y 30 segundos. 

In [2]:
xlsx = pd.ExcelFile("/content/U382_BASE2.xlsx")
df = pd.read_excel(xlsx, "CONSULTA1_AGR_F_APERTURA" )

In [3]:
df.head()

Orden_Corta  Recuento F_APERTURA  SUMA_RECAMBIOS  SUMA_MO  \
0       775439         1 2015-01-02            0.00    19.83   
1       774412         1 2015-01-02          119.78    62.04   
2       774401         1 2015-01-02          136.80   161.70   
3       774715         2 2015-01-02          236.36   317.02   
4       775595         1 2015-01-02            0.00     0.00   

   SUMA_HORAS_FACTURADAS  AÑOS_VEHICULO  
0                   0.28       0.758904  
1                   1.20       2.512329  
2                   2.10       3.306849  
3                   4.20       5.512329  
4                   0.00       5.183562

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 145695 entries, 0 to 145694
Data columns (total 7 columns):
 #   Column                 Non-Null Count   Dtype         
---  ------                 --------------   -----         
 0   Orden_Corta            145695 non-null  int64         
 1   Recuento               145695 non-null  int64         
 2   F_APERTURA             145695 non-null  datetime64[ns]
 3   SUMA_RECAMBIOS         145695 non-null  float64       
 4   SUMA_MO                145694 non-null  float64       
 5   SUMA_HORAS_FACTURADAS  145695 non-null  float64       
 6   AÑOS_VEHICULO          145695 non-null  float64       
dtypes: datetime64[ns](1), float64(4), int64(2)
memory usage: 7.8 MB


In [5]:
df.describe()

Orden_Corta       Recuento  SUMA_RECAMBIOS        SUMA_MO  \
count  1.456950e+05  145695.000000   145695.000000  145694.000000   
mean   1.070782e+06       1.453564      334.529579     242.391004   
std    1.727761e+05       0.850063      702.359577     371.963158   
min    3.532300e+05       1.000000   -16621.630000  -16621.630000   
25%    9.238030e+05       1.000000       42.120000      60.200000   
50%    1.063734e+06       1.000000      167.760000     139.565000   
75%    1.220650e+06       2.000000      383.900000     279.560000   
max    1.377125e+06      16.000000    27892.830000   33243.260000   

       SUMA_HORAS_FACTURADAS  AÑOS_VEHICULO  
count          145695.000000  145695.000000  
mean                4.443828       5.785898  
std                 7.411066       4.488056  
min              -313.620000       0.000000  
25%                 1.000000       2.167123  
50%                 2.200000       4.410959  
75%                 4.650000       8.369863  
max               763.860000      29.953425

In [6]:
df.isnull().sum()

Orden_Corta              0
Recuento                 0
F_APERTURA               0
SUMA_RECAMBIOS           0
SUMA_MO                  1
SUMA_HORAS_FACTURADAS    0
AÑOS_VEHICULO            0
dtype: int64

#HACEMOS EL CLUSTER

In [7]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [8]:
from google.colab import files # Para manejar los archivos y, por ejemplo, exportar a su navegador
import glob # Para manejar los archivos y, por ejemplo, exportar a su navegador
from google.colab import drive # Montar tu Google drive

In [9]:
# Import Libraries

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

from sklearn.preprocessing import StandardScaler
from sklearn.cluster import KMeans

In [10]:
# Suppress Warning

import warnings
warnings.filterwarnings("ignore")

In [11]:
df.head()

Orden_Corta  Recuento F_APERTURA  SUMA_RECAMBIOS  SUMA_MO  \
0       775439         1 2015-01-02            0.00    19.83   
1       774412         1 2015-01-02          119.78    62.04   
2       774401         1 2015-01-02          136.80   161.70   
3       774715         2 2015-01-02          236.36   317.02   
4       775595         1 2015-01-02            0.00     0.00   

   SUMA_HORAS_FACTURADAS  AÑOS_VEHICULO  
0                   0.28       0.758904  
1                   1.20       2.512329  
2                   2.10       3.306849  
3                   4.20       5.512329  
4                   0.00       5.183562

In [14]:
df.info() # tengo que cambiar formatos. 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 145695 entries, 0 to 145694
Data columns (total 7 columns):
 #   Column                 Non-Null Count   Dtype         
---  ------                 --------------   -----         
 0   Orden_Corta            145695 non-null  object        
 1   Recuento               145695 non-null  int64         
 2   F_APERTURA             145695 non-null  datetime64[ns]
 3   SUMA_RECAMBIOS         145695 non-null  float64       
 4   SUMA_MO                145694 non-null  float64       
 5   SUMA_HORAS_FACTURADAS  145695 non-null  float64       
 6   AÑOS_VEHICULO          145695 non-null  float64       
dtypes: datetime64[ns](1), float64(4), int64(1), object(1)
memory usage: 7.8+ MB


In [15]:
df["Orden_Corta"] = df["Orden_Corta"].astype(str)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 145695 entries, 0 to 145694
Data columns (total 7 columns):
 #   Column                 Non-Null Count   Dtype         
---  ------                 --------------   -----         
 0   Orden_Corta            145695 non-null  object        
 1   Recuento               145695 non-null  int64         
 2   F_APERTURA             145695 non-null  datetime64[ns]
 3   SUMA_RECAMBIOS         145695 non-null  float64       
 4   SUMA_MO                145694 non-null  float64       
 5   SUMA_HORAS_FACTURADAS  145695 non-null  float64       
 6   AÑOS_VEHICULO          145695 non-null  float64       
dtypes: datetime64[ns](1), float64(4), int64(1), object(1)
memory usage: 7.8+ MB


In [17]:
df.shape # tenemos 145695 pasos de taller y 7 variables

(145695, 7)

In [18]:
df.describe()

Recuento  SUMA_RECAMBIOS        SUMA_MO  SUMA_HORAS_FACTURADAS  \
count  145695.000000   145695.000000  145694.000000          145695.000000   
mean        1.453564      334.529579     242.391004               4.443828   
std         0.850063      702.359577     371.963158               7.411066   
min         1.000000   -16621.630000  -16621.630000            -313.620000   
25%         1.000000       42.120000      60.200000               1.000000   
50%         1.000000      167.760000     139.565000               2.200000   
75%         2.000000      383.900000     279.560000               4.650000   
max        16.000000    27892.830000   33243.260000             763.860000   

       AÑOS_VEHICULO  
count  145695.000000  
mean        5.785898  
std         4.488056  
min         0.000000  
25%         2.167123  
50%         4.410959  
75%         8.369863  
max        29.953425

- La media de Recambios por paso es 334 y la mediana 167.76
- la media de Mo por paso es de 242.39 y la mediana 139.56
- la media de horas por paso es de 4.43 y la mediana 2.20
- la media de la edad de los vehículo es de 5.75 y la mediana 4.41

In [19]:
df.isnull().sum()

Orden_Corta              0
Recuento                 0
F_APERTURA               0
SUMA_RECAMBIOS           0
SUMA_MO                  1
SUMA_HORAS_FACTURADAS    0
AÑOS_VEHICULO            0
dtype: int64

## CHEQUEANDO OUTLIERS

In [ ]:
plt.figure(figsize=(15,5)) # aqui poenmos el tamaño de cada cuadradro. 

plt.subplot(1,2,1)
sns.boxplot(data=df, y="AÑOS_VEHICULO") # este va a ocupar el primer lugar de la segunda fila. 

plt.subplot(1,2,2)
sns.boxplot(data=dataset, y="Km") # este va a ocupar el segundo lugar de la primera fila. 

plt.show()
# lo que estamos viendo es un diagrama de bigotes donde apreciamos que no hay outliers.

# lo voy a hacer con outliers, mas adelante lo veré quitandolos. 